codec/src/main/java/io/netty/handler/codec/compression/JdkZlibDecoder.java

```diff
@@ -207,7 +217,15 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
             if (readFooter) {
                 gzipState = GzipState.FOOTER_START;
                 if (readGZIPFooter(in)) {
-                    finished = true;
+                    finished = !decompressConcatenated;
+
+                    if (finished) {
+                        inflater.end();
```

Comment:
It's really easy to have a slow native resource leak with Inflater if you don't close the inflater.  Make sure you close after the last stream.

Reply:
its already done

Reply:
What's funny is that it wasn't closed before.

Reply:
it was closed in `handlerRemoved` or I am missing anything ?

Reply:
It was, I missed it. As a result, I've no removed the above `iniflater.end`, and the `else` block.